kubernetes/istio.go

```diff
@@ -248,6 +252,64 @@ func getStatus(statuses map[string][]byte) ([]*ProxyStatus, error) {
 	return fullStatus, nil
 }
 
+func (in *K8SClient) GetConfigDump(namespace, podName string) (*ConfigDump, error) {
+	// Fetching the config_dump data, raw.
+	resp, err := in.EnvoyForward(namespace, podName, "/config_dump")
+	if err != nil {
+		log.Errorf("Error fetching config_map: %v", err)
+		return nil, err
+	}
+
+	cd := &ConfigDump{}
+	err = json.Unmarshal(resp, cd)
+	if err != nil {
+		log.Errorf("Error Unmarshalling the config_dump: %v", err)
+	}
+
+	return cd, err
+}
+
+func (in *K8SClient) EnvoyForward(namespace, podName, path string) ([]byte, error) {
+	writer := new(bytes.Buffer)
+
+	clientConfig, err := ConfigClient()
+	if err != nil {
+		log.Errorf("Error getting Kubernetes Client config: %v", err)
+		return nil, err
+	}
+
+	// First try whether the pod exist or not
+	_, err = in.GetPod(namespace, podName)
+	if err != nil {
+		log.Errorf("Couldn't fetch the Pod: %v", err)
+		return nil, err
+	}
+
+	// Create a Port Forwarder
+	portMap := fmt.Sprintf("%d:15000", config.Get().ExternalServices.Istio.EnvoyAdminLocalPort)
+	f, err := config_dump.NewPortForwarder(in.k8s.CoreV1().RESTClient(), clientConfig,
+		namespace, podName, "localhost", portMap, writer)
+	if err != nil {
+		return nil, err
+	}
+
+	// Start the forwarding
+	if err := f.Start(); err != nil {
+		return nil, err
+	}
+
+	// Defering the finish of the port-forwarding
+	defer f.Stop()
+
+	// Ready to create a request
+	resp, code, err := httputil.HttpGet(fmt.Sprintf("http://localhost:15000%s", path), nil, 10*time.Second)
```

Comment:
But here, I guess this query happens in Kiali container, so that 15000 shouldn't be the same from the EnvoyAdminLocalPort ?